sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBuffer.scala

```diff
@@ -139,24 +138,23 @@ trait RapidsBuffer extends AutoCloseable {
  */
 sealed class DegenerateRapidsBuffer(
     override val id: RapidsBufferId,
-    override val meta: TableMeta) extends RapidsBuffer {
+    override val meta: TableMeta) extends RapidsBuffer with Arm {
   override val size: Long = 0L
   override val storageTier: StorageTier = StorageTier.DEVICE
 
   override def getColumnarBatch(sparkTypes: Array[DataType]): ColumnarBatch = {
     val rowCount = meta.rowCount
-    val nullCount = Optional.of(java.lang.Long.valueOf(0))
-    val columnMeta = new ColumnMeta
-    val columns = new Array[ColumnVector](meta.columnMetasLength)
-    (0 until meta.columnMetasLength).foreach { i =>
-      meta.columnMetas(columnMeta, i)
-      assert(columnMeta.childrenLength == 0, "child columns are not yet supported")
-      val dtype = DType.fromNative(columnMeta.dtypeId(), columnMeta.dtypeScale())
-      columns(i) = GpuColumnVector.from(new ai.rapids.cudf.ColumnVector(
-        dtype, rowCount, nullCount, null, null, null),
-        sparkTypes(i))
+    val packedMeta = meta.packedMetaAsByteBuffer()
+    if (packedMeta != null) {
+      withResource(DeviceMemoryBuffer.allocate(0)) { deviceBuffer =>
```

Comment:
I am curious if there is an advantage to having `DeviceMemoryBuffer` be non-null in this case. I assume it makes some things easier => we can slice it/add ref count, but then it's extra objects. I am not saying there's anything wrong, just mostly wondering in the mostly degenerate case (some kind of `count`), whether we are better off not having the `DeviceMemoryBuffers` around.

Reply:
IIRC I think it's because I couldn't generate the packed metadata without a valid device buffer.

> mostly wondering in the mostly degenerate case (some kind of count), whether we are better off not having the DeviceMemoryBuffers around

For the `count` case we won't be creating the degenerate device buffer because that's a case where there are zero columns, only rows, and libcudf doesn't support tables without columns.  This code path would only be used for the case where there's a table with columns but no rows.